modules/core/src/core/lib/deck.js

```diff
@@ -304,7 +304,8 @@ export default class Deck {
     // Note: avoid React setState due GL animation loop / setState timing issue
     this.layerManager = new LayerManager(gl, {
       eventManager: new EventManager(canvas),
-      stats: this.stats
+      stats: this.stats,
+      useDevicePixels: this.props.useDevicePixels
```

Comment:
I wonder why this.setProps on line 317 does not take care of this case?

Reply:
cause the layerManager is not created at that time.

Reply:
This is the wrong place to handle the prop. `useDevicePixels` may be changed after layerManager is created.
`Deck` calls `layerManager.setParameters(newProps);` in `setProps`, and `setParameters` should update its context with the latest `useDevicePixels`.

Reply:
There is a difference between how props are handled in React and Pure JS. 

In React, the props are always merged with default values, so all props get set every time, even if they were not supplied.

In Pure JS however, props only get set if they were supplied.

If the different components (Deck, LayerManager, AnimationLoop) have different defaults it could be a reason that an explicit mention of the prop is required.

Reply:
This prop is optional. Deck calls `this.setProps(props);` in its constructor where `props` is user props merged with defaults.
We handle all other shared props between Deck/LayerManager in `setParameters` and this one should not be different.